Updated Documentation For Change in Behaviour of Reading X-Ray Tracing Info From Lambda (#27)
open-telemetry/semantic-conventions: specification/trace/semantic_conventions/instrumentation/aws-lambda.md

### AWS X-Ray Environment Span Link

There are two common ways to read the tracing information if AWS X-Ray is enabled for a lambda function:

> 1. Via the system property `com.amazonaws.xray.traceHeader` in the event of a Java function.

Author: Hi Tyler, is this any better? If not, could you please offer a suggested alternative? I'm trying to help the community but seem to be getting the run around somewhat on this.

### AWS X-Ray Environment Span Link

If the `_X_AMZN_TRACE_ID` environment variable is set, instrumentation SHOULD try to parse an
OpenTelemetry `Context` out of it using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md). If the
resulting `Context` is [valid](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isvalid) then a [Span Link][] SHOULD be added to the new Span's
[start options](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#specifying-links) with an associated attribute of `source=x-ray-env` to
indicate the source of the linked span.
Instrumentation MUST check if the context is valid before using it because the `_X_AMZN_TRACE_ID` environment variable can
contain an incomplete trace context which indicates X-Ray isn’t enabled. The environment variable will be set and the
`Context` will be valid and sampled only if AWS X-Ray has been enabled for the Lambda function. A user can
disable AWS X-Ray for the function if the X-Ray Span Link is not desired.

> **Note**: If you are instrumenting a Java AWS lambda then instrumentation will try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) first before checking and attempting to parse the environment variable above.

Author: Hi Christian, there is no advantage.  This is an alternative way to achieve the same thing.  This was opened to address this: 

https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7579

Long story short - you need hacky solutions to set the environment variable and its now highly discouraged by Oracle in newer versions of Java.

> **Note**: If you are instrumenting a Java AWS lambda then instrumentation will try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) first before checking and attempting to parse the environment variable above.

Reviewer: Lambda Java 17 Runtime and further will only use system properties approach. That's the main advantage, to make OTEL work with Lambda properly

`Context` will be valid and sampled only if AWS X-Ray has been enabled for the Lambda function. A user can
disable AWS X-Ray for the function if the X-Ray Span Link is not desired.

> **Note**: If you are instrumenting a Java AWS lambda then instrumentation will try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) first before checking and attempting to parse the environment variable above.

Reviewer 2: ```suggestion
**Note**: When instrumenting a Java AWS Lambda then instrumentation SHOULD first try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) before checking and attempting to parse the environment variable above.
```

> **Note**: When instrumenting a Java AWS Lambda, instrumentation SHOULD first try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) before checking and attempting to parse the environment variable above.
> As above, if the resulting `Context` is valid then a Span Link SHOULD be added to the new Spans start options.

Reviewer 3: ```suggestion
**Note**: When instrumenting a Java AWS Lambda, instrumentation SHOULD first try to parse an OpenTelemetry `Context` out of the system property `com.amazonaws.xray.traceHeader` using the [AWS X-Ray Propagator](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md) before checking and attempting to parse the environment variable above.
```
